src/shared/nested-dropdown-list/list-item.tsx

```diff
@@ -0,0 +1,209 @@
+import { useCallback, useRef, useState } from "react";
+import { find, isEqual } from "lodash";
+import { FItemOption, Mode, TruncateType } from "./types";
+import { StringHelper } from "../../util";
+import {
+    ArrowButton,
+    Category,
+    Label,
+    List,
+    ListItemSelector,
+    Title,
+    TriangleIcon,
+    TruncateContainer,
+    TruncateFirstLine,
+    TruncateSecondLine,
+} from "./list-item.styles";
+
+interface ListItemProps<V1, V2, V3> {
+    item: FItemOption<V1, V2, V3>;
+    mode: Mode;
+    parentKeys: string[];
+    listExtractor?: ((item: FItemOption<V1, V2, V3>) => string) | undefined;
+    selectedItems?: (V1 | V2 | V3)[] | undefined;
+    selectedKeys: string[];
+    defaultExpandKeys: string[];
+    itemTruncationType?: TruncateType | undefined;
+    visible: boolean;
+    onSelect: (item: FItemOption<V1, V2, V3>) => void;
+    onExpand: () => void;
+    onBlur: () => void;
+}
+
+export const ListItem = <V1, V2, V3>({
+    item,
+    mode,
+    parentKeys,
```

Comment:
each formatted item has `categories` which track the current path, the parentKeys seems redundant